src/main/java/org/ays/auth/service/impl/AysUserUpdateServiceImpl.java

```diff
@@ -54,9 +54,13 @@ public void update(final String id,
                        final AysUserUpdateRequest updateRequest) {
 
         final AysUser user = userReadPort.findById(id)
-                .filter(userFromDatabase -> identity.getInstitutionId().equals(userFromDatabase.getInstitution().getId()))
                 .orElseThrow(() -> new AysUserNotExistByIdException(id));
 
+        String institutionId = identity.getInstitutionId();
+        if (!institutionId.equals(user.getInstitution().getId())) {
+            throw new AysUserNotExistByIdException(id);
+        }
```

Comment:
Aşağıdaki gibi daha iyi olur gibi düşünüyorum. Ekstra bir if yazmamış oluruz
```suggestion
        final String institutionId = identity.getInstitutionId();
        
        final AysUser user = userReadPort.findById(id)
                .filter(userFromDatabase -> identity.getInstitutionId().equals(userFromDatabase.getInstitution().getId()))
                .orElseThrow(() -> new AysUserNotExistByIdException(id));
```

Reply:
@agitrubard  yukarıda da bahsettim ama görmedin sanırım 
findById Optional.empty() döndürdüğünde identity.getInstitutionId() çağrısı yapılmakta bu yüzden bazı testler fail oluyor.
bu yüzden bu şekilde ayırmak zorunda kaldım.

Reply:
@MenekseYuncu o yorum gözümden kaçmış, tamamdır, eline sağlık ✅